lib/Logger/LogAppender.cpp

```diff
@@ -233,6 +234,15 @@ void LogAppender::reopen() {
   LogAppenderFile::reopenAll();
 }
 
+void LogAppender::logMessageGuarded(LogMessage const& message) {
+  // Only one thread is allowed to actually write logs to the file.
+  // We use a recusive lock here, just in case writing the log message
+  // causes a crash, in this case we may trigger another force-direct
+  // log. This is not very likely, but it is better to be safe than sorry.
+  RECURSIVE_WRITE_LOCKER(_logOutputMutex, _logOutputMutexOwner);
```

Comment:
Not sure if it is possible, but can we have two different appenders writing to the same sink, e.g. stdout?
In this case, the previous implementation serialized the writes to the sink, and with this PR they can execute concurrently, leading to potentially garbled output.
But we should first check if it is actually possible to have two different appenders write to the same output file, fd, sink, ...

Reply:
Good Point.
This is indeed possible.
Will apply our discussed idea

Reply:
Applied, we should now only have a single Appender per File